### PR TITLE
Restyle the setup task list to match new designs

### LINKED ISSUE #5243: Home Screen - consolidate the task UI
**Description:** we aim to consolidate some of the tasks' elements, in order to provide a cohesive experience when the user is completing the tasks displayed in the task list.

**Acceptance criteria:**
- [ ] Every task area needs to have a title, counter with the number of tasks, and an option to show/hide all the tasks
- [ ] The setup task area is the only one that can be dismissed
- [ ] Instead of showing the progress bar in the setup task area we should display the counter with the number of incomplete tasks
- [ ] The rest of the setup tasks UI can remain the same in this iteration: completed tasks checkmark, dismiss feature, etc
- [ ] The tasks should have the following order: setup, orders, stock, reviews, 3PDs

**Event tracking:**
NA

**Additional links:**
[P2 with links to the Figma file](pbIJXs-ma-p2)

### PR DESCRIPTION
**Pull request by samueljseay**
Partially fixes #5243 

For the first iteration of the home screen redesign, refactoring of the setup task list is kept to a minimum but includes a rework of the setup task list header. Instead of showing the progress bar in the setup task area it now displays a counter with the number of incomplete tasks.

**Note that the rest of the setup tasks UI remain the same in this iteration: completed tasks checkmark, dismiss feature, etc**

As part of this I have promoted the counter badge from Accordion to its own component because its very reusable in a bunch of different contexts.

### Screenshots

Screenshot 2020-11-03 at 3 26 43 PM

### Detailed test instructions:

1. Check that the Accordion component still looks correct in Storybook. To run storybook: `npm i && npm run storybook`
2. On a site with incomplete setup tasks, inspect the visual changes to the setup header. Notes for the latest design iteration [here](https://github.com/woocommerce/woocommerce-admin/issues/5243#issuecomment-715459610)

### Changelog Note:

Dev: Add the new `<Badge>` component to `@woocommerce/components`
Dev: Restyle the setup task list header to display incomplete tasks

### CODE REVIEW

File: client/dashboard/style.scss [hunk 1]
```diff
@@ -99,27 +99,3 @@
 		}
 	}
 }
-
-.woocommerce-dashboard-card {
```
Comment (samueljseay) [inline client/dashboard/style.scss:103]: Removed the only use of this

File: client/task-list/style.scss [hunk 2]
```diff
@@ -489,48 +494,6 @@
 	}
 }
 
-.woocommerce-task-dashboard__container .woocommerce-task-card__progress-bar {
```
Comment (samueljseay) [inline client/task-list/style.scss:492]: Removed the only use of this

File: packages/components/src/accordion/style.scss [hunk 3]
```diff
@@ -1,16 +1,5 @@
...
-		text-align: center;
-		font-style: normal;
-		font-weight: 600;
-		font-size: 14px;
-		line-height: 27px;
-		align-items: center;
-		width: 32px;
-		height: 28px;
+	.woocommerce-badge {
 		margin-left: $gap;
```
Comment (samueljseay) [inline packages/components/src/accordion/style.scss:3]: I've kept style related to the external layout in the accordion.

**Comment by becdetat** [comment]
Looks great :-) 🚢

**Review by becdetat — APPROVED** [review]

**Pull request merged**